src/fs.h

```diff
@@ -155,6 +155,18 @@ static inline path PathFromString(const std::string& string)
 #endif
 }
 
+/**
+ * Convert byte string to path object which in turn has been
+ * normalized, with redundant "." and ".." path components
+ * and trailing separators removed.
+ */
+static inline path NormalizedPathFromString(std::string string)
```

Comment:
In commit "util, refactor: Add fs::NormalizedPathFromString function" (bc9ea0d3506610c0613bc27ee0c5c9d56e91092a)

I think it would probably be better not to add this function because:

- We already have other functions for converting strings to paths (`PathFromString` and `u8path`), and introducing a third function will probably add confusion. Especially when there is no explanation about when this function is supposed to be favored or avoided compared to the other ones.
- I don't think as a general rule that it is a good thing to normalize paths. I think debugging and understanding log messages if easier if paths are passed around verbatim and manipulated as little as possible, and the operating system is allowed to interpret them.
- If it's useful to have a normalization function that removes trailing slashes, I don't think there's a reason for it to do string decoding at the same. This function would be simpler and more accessible as:

```c++
static inline path NormalizedPath(std::path path)
{
    fs::path result = path.lexically_normal();
    // Remove trailing slash, if present.
    return result.has_filename() ? result : result.parent_path();
}
```

Reply:
The commit has been dropped.